File: src/main/java/net/atos/zac/documenten/model/InboxDocumentListParameters.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.documenten.model;

import net.atos.zac.shared.model.ListParameters;
import net.atos.zac.zoeken.model.DatumRange;


public class InboxDocumentListParameters extends ListParameters {

    private String titel;

    private String identificatie;

    private DatumRange creatiedatum;

    public InboxDocumentListParameters() {
        super();
    }

    public String getTitel() {
        return titel;
    }

    public void setTitel(final String titel) {
        this.titel = titel;
    }

    public String getIdentificatie() {
        return identificatie;
    }

    public void setIdentificatie(final String identificatie) {
        this.identificatie = identificatie;
    }

    public DatumRange getCreatiedatum() {
        return creatiedatum;
    }

    public void setCreatiedatum(final DatumRange creatiedatum) {
        this.creatiedatum = creatiedatum;
    }
}

File: src/main/app/src/app/documenten/model/inbox-document-list-parameters.ts
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

import { SortDirection } from "@angular/material/sort";
import { ListParameters } from "../../shared/model/list-parameters";
import { DatumRange } from "../../zoeken/model/datum-range";
import { ZoekFilters } from "../../gebruikersvoorkeuren/zoekopdracht/zoekopdracht.component";

export class InboxDocumentListParameters
  extends ListParameters
  implements ZoekFilters
{
  readonly filtersType = "InboxDocumentListParameters";
  identificatie: string;
  creatiedatum = new DatumRange();
  titel: string;

  constructor(sort: string, order: SortDirection) {
    super(sort, order);
  }

  static heeftActieveFilters(zoekFilters: any): boolean {
    if (zoekFilters.identificatie != null) {
      return true;
    }
    if (
      zoekFilters.creatiedatum?.van != null ||
      zoekFilters.creatiedatum?.tot != null
    ) {
      return true;
    }
    if (zoekFilters.titel != null) {
      return true;
    }
    return false;
  }
}
